### PR TITLE
Fixed Gutenberg block

### PR DESCRIPTION
**Pull request by HardeepAsrani**
Kept up with Gutenberg updates, which fixes:
https://github.com/Codeinwp/cf7-gutenberg/issues/5 and
https://github.com/Codeinwp/cf7-gutenberg/issues/6

### CODE REVIEW

**Review by andreilupu — APPROVED** [review]
👍 Good to go!

File: class-cf7-gutenberg-block-plugin.php [hunk 1]
```diff
@@ -23,7 +23,7 @@ function gutenberg_enqueue_block_editor_assets() {
 		wp_enqueue_script(
 			'gutenberg-cf7',
 			plugins_url( '/block-esnext/block.build.js', __FILE__ ),
-			array( 'wp-blocks', 'wp-i18n', 'wp-element', 'underscore' ),
+			array( 'wp-blocks', 'wp-components', 'wp-editor', 'wp-i18n', 'wp-element', 'underscore' ),
```
Comment (ajitbohra) [inline class-cf7-gutenberg-block-plugin.php]: @HardeepAsrani curious to know about the `underscore` dependecy

File: block-esnext/components/Editor.js [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
+	withAPIData,
+} = wp.components;
 
-const {InspectorControls} = wp.blocks;
-const {TextControl, SelectControl} = InspectorControls;
+const { __ } = wp.i18n;
+
+const { InspectorControls } = wp.editor;
 
 class Editor extends Component {
```
Comment (ajitbohra) [inline block-esnext/components/Editor.js:18]: @HardeepAsrani we can convert this to functional component

File: class-cf7-gutenberg-block-plugin.php (hunk 1 above)
Comment (HardeepAsrani) [reply class-cf7-gutenberg-block-plugin.php]: Don't know. It was there before, so didn't touch.

File: block-esnext/components/Editor.js (hunk 2 above)
Comment (HardeepAsrani) [reply block-esnext/components/Editor.js:18]: Thought the same, but left it for this version. :)

**Comment by dillix** [comment]
When will this request be merged to Codeinwp:master?

**Comment by HardeepAsrani** [comment]
@rodica-andronache Can we merge this?

**Comment by rodica-andronache** [comment]
@HardeepAsrani you can merge this if you feel it's ok 👍

File: block-esnext/components/Editor.js [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
  */
-const {Component} = wp.element;
-const {Placeholder, withAPIData} = wp.components;
-const {__} = wp.i18n;
+const { Component } = wp.element;
+const {
+	TextControl,
+	SelectControl,
+	Placeholder,
+	withAPIData,
```
Comment (ajitbohra) [inline block-esnext/components/Editor.js]: @HardeepAsrani  withAPIData is deprecated & removed https://wordpress.org/gutenberg/handbook/reference/deprecated/#3-7-0

Comment (HardeepAsrani) [reply block-esnext/components/Editor.js]: Yea, will have to redo it to see what else has moved.

Comment (ajitbohra) [reply block-esnext/components/Editor.js]: so far this one looks out of place & other is `wp-blocks` dependency in styles which is not needed https://github.com/Codeinwp/cf7-gutenberg/blob/master/class-cf7-gutenberg-block-plugin.php#L35.

Also might be good to use `register_block_type()` for registering assets https://wordpress.org/gutenberg/handbook/blocks/writing-your-first-block-type/#enqueuing-block-scripts

**Pull request merged**